src/Microsoft.ML.Core/Utilities/PathUtils.cs

```diff
@@ -139,5 +141,65 @@ public static string CreateFolderIfNotExists(string folder)
 
             return null;
         }
+
+        /// <summary>
+        /// Make a full path realtive to a base path.
+        /// </summary>
+        /// <param name="basepath">The base path, assumed to be a directory.</param>
+        /// <param name="path">The full path.</param>
+        /// <returns>The relative path.</returns>
+        /// <exception cref="ArgumentException">If the paths are not relative.</exception>
+        public static string MakePathRelative(string basepath, string path)
```

Comment:
>MakePathRelative [](start = 29, length = 16)

We might prefer to have at least two distinct usages of some facility before it is "promoted" to utilities, for two practical reasons:

1. If there's only usage from one assembly, it could have been an internal method and not increased our public surface.

2. Adding it to the public surface places a bit more emphasis on making sure that it is "right." My confidence that we got the "right" abstraction for any hypothetical future usage is somewhat low.

What do we think about keeping this internal so that it can be used from `PartitionedFileLoader` and `Parser`, but no further?

Reply:
I have no object to this, however I can't just make the methods internal since the usage is in a different assembly. So I had to move it to a different file to be internal.

Reply:
Sure, right, that's what I meant, thanks Tyler

---
In reply to: [190063880](https://github.com/dotnet/machinelearning/pull/61#discussion_r190063880) [](ancestors = 190063880)

Reply:
resolved.